pkgs/development/libraries/nvidia-thrust/default.nix

```diff
@@ -0,0 +1,91 @@
+{ lib
+, config
+, fetchFromGitHub
+, stdenv
+, cmake
+, pkg-config
+, cudaSupport ? config.cudaSupport or false
+, cudaPackages
+, symlinkJoin
+, enableOpenmp ? true
+, enableTbb ? false
+, tbb
+, hostSystem ? "CPP"
```

Comment:
What is the difference between `hostSystem` and `deviceSystem`? What do each of the values mean?

Reply:
Thrust exposes `host` and `device` counterparts of structures such as host_vector and device_vector, and the HOST_SYSTEM/DEVICE_SYSTEM macros select their respective backends.

I don't think it makes sense for these to be package arguments though, because thrust is header-only and the backend is chosen at compile-time using macros.

Reply:
Skimming quickly through the docs, they have containers (e.g. vector) for the host memory, and containers for wherever accelerated computation is happening. The former is always CPU, but there are different execution policies (CPP=serial, openmp/tbb=parallel). The latter can run on CPU with the same policies, or it can run on an nvidia GPU ("CUDA")

Reply:
I'll add a comment